cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1123,7 +1160,10 @@ func (p *BicepProvider) createOutputParameters(
 		if found {
 			paramName = canonicalCasing
 		} else {
-			paramName = key
+			// To support BYOI (bring your own infrastructure) scenarios we will default to UPPER when canonical casing
+			// is not found in the parameters file to workaround strange azure behavior with OUTPUT values that look
+			// like `azurE_RESOURCE_GROUP`
+			paramName = strings.ToUpper(key)
```

Comment:
Do we know the percentage of outputs where this is being impacted? Is it 100%? Because performing any type of casing conversion itself can be seen as a bug (unless it is always case-insensitive), so I wonder if we end up impacting all users negatively instead of a small subset of users, where we can absolutely raise the issue with service teams as users start using this feature.

By default, this ends up making canonical bicep output names like `myValueThatIsALongName`  showing up as `MYVALUETHATISALONGNAME`.

Reply:
If you have an infra bicep template, then the casing is preserved from whatever `output` variables you described in your main module.

This only applies in the case where we do not have any input template to map against and just default to UPPER.